content/tokio/topics/tracing.md

```diff
@@ -0,0 +1,122 @@
+data as well as textual messages.
+
+You can use `tracing` to:
+- emit distributed traces to an [OpenTelemetry] collector
+- debug your application with [Tokio Console]
+- log to [`stdout`], [a log file] or [`journald`]
+- [profile] where your application is spending time
+
+[`tracing`]: https://docs.rs/tracing
+[`tracing-subscriber`]: https://docs.rs/tracing-subscriber
+[OpenTelemetry]: https://docs.rs/tracing-opentelemetry
+[Tokio Console]: https://docs.rs/console-subscriber
+[`stdout`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html
+[a log file]: https://docs.rs/tracing-appender/latest/tracing_appender/
+[`journald`]: https://docs.rs/tracing-journald/latest/tracing_journald/
+[profile]: https://docs.rs/tracing-timing/latest/tracing_timing/
+
+# Setup
+
+To begin, add [`tracing`] and [`tracing-subscriber`] as dependencies:
+
+```toml
+[dependencies]
+tracing = "0.1.34"
+tracing-subscriber = "0.3"
+```
+
+The [`tracing`] crate provides the API we will use to emit traces. The 
+[`tracing-subscriber`] crate provides some basic utilities for forwarding those 
+traces to external listeners (e.g., `stdout`).
+
+# Subscribing to Traces
+
+If you are authoring an executable (as opposed to a library), you will need to 
+register a tracing *subscriber*. Subscribers are types that process traces 
+emitted by your application and its dependencies, and and can perform tasks 
+such as computing metrics, monitoring for errors, and re-emitting traces to the 
+outside world (e.g., `journald`, `stdout`, or an `open-telemetry` daemon).
+
+In most circumstances, you should register your tracing subscriber as early as
+possible in your `main` function. For instance, the [`FmtSubscriber`] type
+provided by [`tracing-subscriber`] prints formatted traces and events to
+`stdout`, can be registered like so:
+
+```rust
+#[tokio::main]
+pub async fn main() -> mini_redis::Result<()> {
+    // construct a subscriber that prints formatted traces to stdout
+    let subscriber = tracing_subscriber::FmtSubscriber::new();
+    // use that subscriber to process traces emitted after this point
+    tracing::subscriber::set_global_default(subscriber)?;
+
+    ...
+}
+```
+
+[`FmtSubscriber`]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html
+
+If you run your application now, you may see some trace events emitted by tokio,
+but you will need to modify your own application to emit traces to get the most
+out of `tracing`.
+
+# Emitting Traces
+
+The easiest way to emit traces is with the [`instrument`] proc-macro annotation
+provided by [`tracing`], which re-writes the bodies of functions to emit traces
+each time they are invoked.
+
+[`instrument`]: https://docs.rs/tracing/latest/tracing/attr.instrument.html
+
+For instance, to trace the method in `mini-redis-server` that handles each
+connection:
+
+```rust
+impl Handler {
+    /// Process a single connection.
+    #[instrument(
+        level = "info",
```

Comment:
Eliza alluded [to this earlier](https://github.com/tokio-rs/website/pull/653/files#r866154130), but I wouldn't start with this example—we already default to `INFO` and this might contribute to overload. I strongly suggest starting simple. It can also be helpful to show the desugared version of this macro in terms of `span!`.

Reply:
i'm fine with not showing the function-like `span!` macro in this documentation...it might be worth just including a link to the API docs that says like "for details on how spans are recorded, see here" or something.